src/pages/top-up/Swap.tsx

```diff
@@ -52,28 +47,65 @@ export function Swap({ header }: Props): ReactElement {
   const navigate = useNavigate()
   const { enqueueSnackbar } = useSnackbar()
 
+  // Fetch current price of BZZ
   useEffect(() => {
     // eslint-disable-next-line no-console
     getBzzPriceAsDai(desktopUrl).then(setPrice).catch(console.error)
   }, [desktopUrl])
 
-  if (!balance || !nodeAddresses) {
-    return <Loading />
-  }
+  // Set the initial xDAI to swap
+  useEffect(() => {
+    if (!balance) {
+      return
+    }
 
-  const optimalSwap = balance.dai.minusBaseUnits('1')
-  const lowAmountSwap = new DaiToken(balance.dai.toBigNumber.dividedToIntegerBy(2))
+    if (balance.dai.toDecimal.isGreaterThanOrEqualTo(DaiToken.fromDecimal('1').plusBaseUnits(MINIMUM_XDAI).toDecimal)) {
+      // Balance has at least 1 + MINIMUM_XDAI xDai
+      setDaiToSwap(balance.dai.minusBaseUnits('1'))
+    } else {
+      // Balance is low, halve the amount
+      setDaiToSwap(new DaiToken(balance.dai.toBigNumber.dividedToIntegerBy(2)))
+    }
+  }, [balance])
 
-  let daiToSwap: DaiToken
+  // Set the xDAI to swap based on user input
+  useEffect(() => {
+    setError(null)
+    try {
+      if (userInputSwap) {
+        const dai = DaiToken.fromDecimal(userInputSwap)
+        setDaiToSwap(dai)
+
+        if (dai.toDecimal.lte(0)) {
+          setError('xDAI to swap must be a positive number')
```

Comment:
Shouldn't there be also check for the minimal xDAI?

Reply:
That is part of the next useEffect:

```
if (daiAfterSwap.toDecimal.lt(MINIMUM_XDAI))
    setError(`Must keep at least ${MINIMUM_XDAI} xDAI after swap!`)
```

<img width="495" alt="image" src="https://user-images.githubusercontent.com/77121044/200571241-58555830-30db-4078-9361-7674f462c07f.png">

Reply:
👍